Content.Client/GameObjects/Components/StackVisualizer.cs

```diff
@@ -0,0 +1,145 @@
+﻿#nullable enable
+
+using System.Collections.Generic;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.Utility;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.Interfaces.GameObjects.Components;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Utility;
+using YamlDotNet.RepresentationModel;
+
+namespace Content.Client.GameObjects.Components
+{
+    /// <summary>
+    /// Visualizer for items that come in stacks and have different appearance
+    /// depending on the size of the stack.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// - type: Appearance
+    ///   visuals:
+    ///     - type: StackVisualizer
+    ///       stackLayers:
+    ///         - goldbar_10
+    ///         - goldbar_20
+    ///         - goldbar_30
+    /// </code>
+    /// </example>
+    [UsedImplicitly]
+    public class StackVisualizer : AppearanceVisualizer
+    {
+        /// <summary>
+        /// Default IconLayer stack.
+        /// </summary>
+        private const int IconLayer = 0;
+        /// <summary>
+        /// Sprite layers used in stack visualizer. Sprites first in layer correspond to lower stack states
+        /// e.g. <code>_spriteLayers[0]</code> is lower stack level than <code>_spriteLayers[1]</code>.
+        /// </summary>
+        private readonly List<string> _spriteLayers = new();
+        /// <summary>
```

Comment:
```suggestion
        /// <summary>
        /// Default IconLayer stack.
        /// </summary>
        private const int IconLayer = 0;
        
        /// <summary>
        /// Sprite layers used in stack visualizer. Sprites first in layer correspond to lower stack states
        /// e.g. <code>_spriteLayers[0]</code> is lower stack level than <code>_spriteLayers[1]</code>.
        /// </summary>
        private readonly List<string> _spriteLayers = new();
        
        /// <summary>
```

```suggestion
        private readonly List<string> _spriteLayers = new();
        
        /// <summary>
```

Just to make the block more readable a space between fields.